src/routing/lazy_messaging.rs

```diff
@@ -6,121 +6,137 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
+use crate::routing::command::Command;
+use crate::routing::core::LaggingMessages;
 use crate::{
-    agreement::Proposal,
     error::Result,
-    messages::{Message, MessageHash, Variant},
+    messages::{Message, Variant},
     network::Network,
     node::Node,
     section::Section,
 };
-use sn_messaging::DstLocation;
+use sn_messaging::{DestInfo, DstLocation};
 use std::cmp::Ordering;
+use std::net::SocketAddr;
 
 /// On reception of an incoming message, determine the actions that need to be taken in order to
-/// bring ours and the senders knowledge about each other up to date.
+/// bring ours and the senders knowledge about each other up to date. Returns a tuple of
+/// `Actions` and `bool`. The boolean is flag for executing the message. If entropy is found, we do
+/// not execute the message by returning `false`.
 pub(crate) fn process(
     node: &Node,
     section: &Section,
     network: &Network,
+    lagging_messages: &mut LaggingMessages,
     msg: &Message,
-) -> Result<Actions> {
+    dest_info: DestInfo,
+    sender: Option<SocketAddr>,
+) -> Result<(Actions, bool)> {
     let mut actions = Actions::default();
-    let mut send_other_section = false;
 
     let src_name = msg.src().name();
     if section.prefix().matches(&src_name) {
         // This message is from our section. We update our members via the `Sync` message which is
         // done elsewhere.
-        return Ok(actions);
+        return Ok((actions, true));
     }
 
-    if let Ok(src_key) = msg.proof_chain_last_key() {
-        if !network.has_key(src_key) && !section.chain().has_key(src_key) {
-            // We don't know the src key. Send them a `OtherSection` message whose `dst_key` is set
-            // to the latest key we know and when they receive it they would send us back a
-            // `OtherSection` with their latest info, including their latest key.
-            //
-            // NOTE: the reason why we update the key only after receiving their `OtherSection` and
-            // not right here is to keep their key and their section_authority_provider in sync.
-            trace!("lazy messaging: src key unknown");
-            send_other_section = true;
+    let dst = msg.src().src_location().to_dst();
+
+    if let Ok(src_chain) = msg.proof_chain() {
```

Comment:
just highlight that here we use the proof_chain from the Message,
meanwhile I remember there is SectionKnowledge, which is payload of Message, contains a separate chain field.
worth to check whether it has been properly used, or can be removed for optimisation.